shai/src/main/scala/at/forsyte/apalache/shai/v1/CmdExecutorService.scala

```diff
@@ -18,6 +35,63 @@ import com.typesafe.scalalogging.Logger
 
 class CmdExecutorService(logger: Logger) extends ZioCmdExecutor.ZCmdExecutor[ZEnv, Any] {
 
-  // TODO These will be used when we start adding the RPC calls
   val _todo = logger
+
+  /** Concurrent tasks performed by the service that produce values of type `T` */
+  type Result[T] = ZIO[ZEnv, Status, T]
+
+  def run(req: CmdRequest): Result[CmdResponse] = for {
+    cmd <- validateCmd(req.cmd)
+    resp <- executeCmd(cmd, req.config) match {
+      case Right(r)  => ZIO.succeed(CmdResponse.Result.Success(r))
+      case Left(err) => ZIO.succeed(CmdResponse.Result.Failure(err))
+    }
+  } yield CmdResponse(resp)
+
+  private def executeCmd(cmd: Cmd, cfgStr: String): Either[String, String] = {
+    // Convert a Try into an `Either` with `Left` the message from a possible `Failure`.
+    def convErr[O](v: Try[O]) = v.toEither.left.map(e => e.getMessage())
+
+    for {
+      cfg <- parseConfig(cfgStr)
+
+      toolModule <- {
+        import OptionGroup._
+        cmd match {
+          case Cmd.PARSE           => convErr(WithIO(cfg)).map(new ParserModule(_))
+          case Cmd.CHECK           => convErr(WithCheckerPreds(cfg)).map(new CheckerModule(_))
+          case Cmd.TYPECHECK       => convErr(WithTypechecker(cfg)).map(new TypeCheckerModule(_))
+          case Cmd.Unrecognized(_) => throw new Exception("invalid: toolModuleOfCmd applied before validateCmd")
```

Comment:
What does `toolModuleOfCmd` refer to? Should it be `executeCmd`?

Reply:
Since this will be propagated to the client-side, it's probably better to just report that an unsupported command has been passed?

Reply:
The user facing error gets handled during validation here: https://github.com/informalsystems/apalache/blob/9440e380656cbc7580ee48a8527b787a9400debf/shai/src/main/scala/at/forsyte/apalache/shai/v1/CmdExecutorService.scala#L78-L85

This case should be impossible. The error message is out of date! Thanks :)

Reply:
Already merged, but can I suggest to `s/Exception/IllegalArgumentException` in this case?

Reply:
Oh good idea. Can add in a followup.